#! /bin/sh /usr/share/dpatch/dpatch-run
## dumpobj.dpatch by Samuel Mimram <smimram@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Ship ocamldumpobj and add a manpage for it.

@DPATCH@
diff -urNad ocaml~/man/ocamlc.m ocaml/man/ocamlc.m
--- ocaml~/man/ocamlc.m	2008-10-20 14:29:27.000000000 +0200
+++ ocaml/man/ocamlc.m	2008-10-20 14:31:41.000000000 +0200
@@ -579,7 +579,8 @@
 Display a short usage summary and exit.
 
 .SH SEE ALSO
-.BR ocamlopt (1), \ ocamlrun (1), \ ocaml (1), \ ocamlobjinfo (1).
+.BR ocamlopt (1), \ ocamlrun (1), \ ocaml (1),
+.BR ocamlobjinfo (1), \ ocamldumpobj (1).
 .br
 .IR "The Objective Caml user's manual" ,
 chapter "Batch compilation".
diff -urNad ocaml~/man/ocamldumpobj.m ocaml/man/ocamldumpobj.m
--- ocaml~/man/ocamldumpobj.m	1970-01-01 01:00:00.000000000 +0100
+++ ocaml/man/ocamldumpobj.m	2008-10-20 14:29:28.000000000 +0200
@@ -0,0 +1,20 @@
+.TH OCAMLDUMPOBJ 1 "January 4, 2006"
+.SH NAME
+ocamldumpobj \- disassembler for OCaml executable and .cmo object files
+.
+.SH SYNOPSIS
+.B ocamldumpobj
+.RI file\ ...
+.
+.SH DESCRIPTION
+Disassembler for executable and .cmo object files compiled by OCaml.
+.
+.SH SEE ALSO
+.BR ocamlc (1),
+.BR ocamlopt (1),
+.BR ocamlobjinfo (1).
+.br
+.
+.SH AUTHOR
+This manual page was written by Samuel Mimram <smimram@debian.org>,
+for the Debian GNU/Linux system (but may be used by others).
diff -urNad ocaml~/man/ocamlopt.m ocaml/man/ocamlopt.m
--- ocaml~/man/ocamlopt.m	2008-10-20 14:29:27.000000000 +0200
+++ ocaml/man/ocamlopt.m	2008-10-20 14:29:28.000000000 +0200
@@ -604,7 +604,8 @@
 
 .SH SEE ALSO
 .BR ocamlc (1),
-.BR ocamlobjinfo (1).
+.BR ocamlobjinfo (1),
+.BR ocamldumpobj (1).
 .br
 .IR "The Objective Caml user's manual" ,
 chapter "Native-code compilation".
diff -urNad ocaml~/tools/Makefile.shared ocaml/tools/Makefile.shared
--- ocaml~/tools/Makefile.shared	2008-10-20 14:29:27.000000000 +0200
+++ ocaml/tools/Makefile.shared	2008-10-20 14:30:25.000000000 +0200
@@ -73,6 +73,7 @@
 	cp ocamlcp $(BINDIR)/ocamlcp$(EXE)
 	cp profiling.cmi profiling.cmo $(LIBDIR)
 	cp objinfo $(BINDIR)/ocamlobjinfo$(EXE)
+	cp dumpobj $(BINDIR)/ocamldumpobj$(EXE)
 
 clean::
 	rm -f ocamlprof ocamlcp
